browser/search/ntp_utils.cc

```diff
@@ -0,0 +1,44 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/search/ntp_utils.h"
+
+#include "brave/browser/profiles/profile_util.h"
+#include "brave/common/pref_names.h"
+#include "chrome/common/pref_names.h"
+#include "components/pref_registry/pref_registry_syncable.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+
+namespace {
+
+void ClearNewTabPageProfilePrefs(Profile* profile) {
+  PrefService* prefs = profile->GetPrefs();
+  prefs->ClearPref(kNewTabPageShowTopSites);
+}
+
+}  // namespace
+
+namespace new_tab_page {
+
+void MigrateNewTabPagePrefs(Profile* profile) {
+  // Migrate over to the Chromium setting for shortcuts visible
+  const PrefService::Preference* pref =
+      profile->GetPrefs()->FindPreference(prefs::kNtpShortcutsVisible);
+  if (!pref->HasUserSetting()) {
+    profile->GetPrefs()->SetBoolean(prefs::kNtpShortcutsVisible,
+      profile->GetPrefs()->GetBoolean(kNewTabPageShowTopSites));
```

Comment:
Does this mean for new users that top sites will default to `false` since the below is settings the old pref to false? 

```
registry->RegisterBooleanPref(kNewTabPageShowTopSites, false);
```

I can't see anything that would only run this code if `kNewTabPageShowTopSites` has already been set.

Reply:
New users shouldn't hit this code path because the migration would only happen if `kNewTabPageShowTopSites` is present in the profile- but let me verify that. I copied what the dark theme migration code was doing

Reply:
Good catch - this logic is definitely wrong ☹️ Looking at proper fix now

Reply:
Fixed! I missed that the example that I copied from didn't change the default value; it just registered it in a new place 😄

Reply:
OK fixed "for real" this time 😄 The value will only ever be set if it was modified by user in the first place

Reply:
makes much better sense!